File: code/csharp/DesignPatterns/builder/Product.cs
namespace Builder;
using System;
using System.Collections.Generic;

class Product
{
    private List<string> _parts = new List<string>();

    public void AddPart(string part)
    {
        _parts.Add(part);
    }

    public void Show()
    {
        Console.WriteLine("Custom Sandwich Ingredients:");
        foreach (string part in _parts)
        {
            Console.WriteLine($"- {part}");
        }
    }
}

File: code/typescript/src/builder/Product.ts
export class Product {
    private parts: string[] = [];

    public addPart(part: string): void {
        this.parts.push(part);
    }

    public show(): void {
        console.log("Custom Sandwich Ingredients:");
        this.parts.forEach(part => {
            console.log(`- ${part}`);
        });
    }
}
